OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/ApplicationControllerTests.cs

```diff
@@ -209,115 +209,127 @@ public async Task GetByParentId_WhenParentHasNoRights_ShouldReturnBadRequest()
+        public async Task GetByPropertyId_WhenIdIsNotValid_ShouldReturnBadRequest(Guid id, string property)
+        {
+            // Act
+            var filter = new ApplicationFilter { Status = (ApplicationStatus)1 };
+
+            var result = await controller.GetByPropertyId(property, id, filter).ConfigureAwait(false) as BadRequestObjectResult;
+
+            // Assert
+            result.Should().NotBeNull();
+            result.StatusCode.Should().Be(400);
+        }
+
+        [Test]
+        [TestCase("83caa2e6-902a-43b5-9744-8a9d66604666", "provider")]
+        [TestCase("f25d1bfc-8ebc-4087-b1c3-8dbb7964222d", "workshop")]
+        public async Task GetByPropertyId_WhenProviderHasNoApplications_ShouldReturnNoContent(Guid id, string property)
+        {
+            // Arrange
+            var filter = new ApplicationFilter { Status = (ApplicationStatus)1 };
+
+            var newProvider = new ProviderDto { Id = new Guid("83caa2e6-902a-43b5-9744-8a9d66604666"), UserId = userId };
+            var newWorkshop = new WorkshopDTO { Id = new Guid("94b81fa7-180f-4965-8aac-908a9f3ecb8d"), ProviderId = new Guid("83caa2e6-902a-43b5-9744-8a9d66604666") };
+
+            httpContext.Setup(c => c.User.IsInRole("provider")).Returns(true);
+
+            providerService.Setup(s => s.GetByUserId(userId)).ReturnsAsync(newProvider);
+            workshopService.Setup(s => s.GetById(id)).ReturnsAsync(newWorkshop);
+            providerService.Setup(s => s.GetById(id)).ReturnsAsync(newProvider);
+            applicationService.Setup(s => s.GetAllByProvider(id, filter))
+                .ReturnsAsync(applications.Where(a => a.Workshop.ProviderId == id));
+            applicationService.Setup(s => s.GetAllByWorkshop(id, filter))
+                .ReturnsAsync(applications.Where(a => a.WorkshopId == id));
+
+            // Act
+            var result = await controller.GetByPropertyId(property, id, filter).ConfigureAwait(false) as NoContentResult;
+
+            // Assert
+            result.Should().NotBeNull();
+            result.StatusCode.Should().Be(204);
+        }
+
+        [Test]
+        [TestCase("83caa2e6-902a-43b5-9744-8a9d66604666", "provider")]
+        [TestCase("f25d1bfc-8ebc-4087-b1c3-8dbb7964222d", "workshop")]
+        public async Task GetByPropertyId_WhenProviderHasNoRights_ShouldReturnBadRequest(Guid id, string property)
+        {
+            // Arrange
+            var filter = new ApplicationFilter { Status = (ApplicationStatus)1 };
+
+            var anotherProvider = new ProviderDto { Id = new Guid("83caa2e6-902a-43b5-9744-8a9d66604777"), UserId = userId };
+
+            httpContext.Setup(c => c.User.IsInRole("provider")).Returns(true);
+            providerService.Setup(s => s.GetByUserId(userId)).ReturnsAsync(anotherProvider);
+            applicationService.Setup(s => s.GetAllByProvider(id, filter))
+                .ReturnsAsync(applications.Where(a => a.Workshop.ProviderId == id));
```

Comment:
Can you please check applications.Where() behaviour because you create applications dynemically without any specific rules and at least it has difference statuses when you call some times (check GetByStatus_WhenStatusIsValid_ShouldReturnOkObjectResult method)?